drivers/SmartThings/wemo/src/discovery.lua

```diff
@@ -42,6 +42,11 @@ local function send_disco_request()
   local sock = assert(socket.udp(), "create discovery socket")
   assert(sock:setsockname(listen_ip, listen_port), "disco| socket setsockname")
   local timeouttime = socket.gettime() + 5 -- 5 second timeout, `MX` + 1 for network delay
+
+  -- TODO Sometimes this is failing because the OS reports the "Network is unreachable", and
+  -- it causes a crash/restart loop. Is this acceptable? Should we stop disco but not crash?
+  -- Some hubs recover when the network becomes available, but some never do, presumably because
+  -- of LAN setups with blocked ports from the hub.
```

Comment:
Will the symptoms here improve once backoff is implemented?

Reply:
My personal preference here is to avoid `assert` as much as possible in the discovery loop and just have it end the iteration early and try again next go-around in the case that a network operation fails since it's a cyclic operation. If the network is messed up, then the hub will simply not be able to discover devices, which seems fine.

Reply:
This seems reasonable. Repeated logs rather than repeated crashes has way less of a chance of having side affects on the hub. I will add though that crashes are our main way of discovering these issues, so if we prevent the crash we will no longer have an idea of the hubs that are experiencing it without gathering user feedback.

Reply:
I pushed a change to log errors to hub logs and end the iteration of the discovery loop should we hit this issue. It will result in many logs to hub logs, but the driver shouldn't crash. Is it worth adding a `socket.sleep(2)` to limit the amount of logs?

Reply:
Note I also made the change for bose.

Reply:
good thinking

Reply:
@dljsjr  Im gonna wait for your approval to merge on this one.